Agent list may include whitespace-only strings - needs investigation

## Problem
The `_list_agents()` function in `health.py` doesn't filter whitespace-only strings:

```python
def _list_agents() -> list[str]:
    result = subprocess.run(...)
    if result.returncode == 0:
        return [
            line.strip() for line in result.stdout.decode().splitlines()
            if line.strip()  # This check is correct
        ]
```

Actually looking at the code, it does check `if line.strip()`. However, the test shows that whitespace-only lines are being included. Let me investigate further.

Actually the issue is that the mock in the test returns `b"   \n   \n"` which when decoded and split gives `["   ", "   "]`, and `line.strip()` on these gives `["", ""]` which should be filtered out.

Wait, looking at the test failure:
```
E   AssertionError: assert ['\n   \n'] == []
```

This suggests the actual output includes `\n   \n` as a single line. The issue is that `.splitlines()` keeps the newline characters, so `line.strip()` should still filter it out.

Let me check the actual implementation again...

Actually the issue might be that `splitlines()` with keepends=False (default) should work. Let me verify the actual behavior.

## Investigation Needed
The test failure suggests that whitespace-only lines are being included in the agents list. Need to investigate why the filtering isn't working.

## Acceptance Criteria
- [ ] Investigate why whitespace lines are included
- [ ] Fix filtering if needed
- [ ] Add tests for edge cases

--- a/src/gasclaw/health.py
+++ b/src/gasclaw/health.py
@@ -88,7 +88,14 @@ def _list_agents() -> list[str]:
             timeout=10,
         )
         if result.returncode == 0:
-            return [line.strip() for line in result.stdout.decode().splitlines() if line.strip()]
+            # Filter out empty lines and whitespace-only lines
+            agents = []
+            for line in result.stdout.decode().splitlines():
+                stripped = line.strip()
+                # Only include non-empty lines (strip() removes all whitespace)
+                if stripped:
+                    agents.append(stripped)
+            return agents
     except (OSError, subprocess.TimeoutExpired):
         pass
     return []

--- a/tests/unit/test_health.py
+++ b/tests/unit/test_health.py
@@ -322,6 +322,84 @@ class TestListAgentsErrorHandling:
         result = _list_agents()
         assert result == []
 
+    def test_list_agents_whitespace_only_lines_filtered(self, monkeypatch):
+        """Test _list_agents filters out whitespace-only lines - Issue #66."""
+        from gasclaw.health import _list_agents
+
+        def mock_run(*a, **kw):
+            # Return output with various whitespace-only lines
+            return subprocess.CompletedProcess(
+                args=a[0] if a else ["cmd"],
+                returncode=0,
+                stdout=b"mayor\n   \n  \t  \n\n\ndeacon\n \n witness\n",
+            )
+
+        monkeypatch.setattr(subprocess, "run", mock_run)
+        result = _list_agents()
+        # Only the actual agent names should be included, not whitespace lines
+        assert result == ["mayor", "deacon", "witness"]
+
+    def test_list_agents_empty_lines_filtered(self, monkeypatch):
+        """Test _list_agents filters out empty lines."""
+        from gasclaw.health import _list_agents
+
+        def mock_run(*a, **kw):
+            return subprocess.CompletedProcess(
+                args=a[0] if a else ["cmd"],
+                returncode=0,
+                stdout=b"agent1\n\nagent2\n\n\nagent3\n",
+            )
+
+        monkeypatch.setattr(subprocess, "run", mock_run)
+        result = _list_agents()
+        assert result == ["agent1", "agent2", "agent3"]
+
+    def test_list_agents_leading_trailing_whitespace_stripped(self, monkeypatch):
+        """Test _list_agents strips leading/trailing whitespace from agent names."""
+        from gasclaw.health import _list_agents
+
+        def mock_run(*a, **kw):
+            return subprocess.CompletedProcess(
+                args=a[0] if a else ["cmd"],
+                returncode=0,
+                stdout=b"  mayor  \n\t deacon \t\n witness  \n",
+            )
+
+        monkeypatch.setattr(subprocess, "run", mock_run)
+        result = _list_agents()
+        assert result == ["mayor", "deacon", "witness"]
+
+    def test_list_agents_all_whitespace_returns_empty(self, monkeypatch):
+        """Test _list_agents returns empty list when all lines are whitespace - Issue #66."""
+        from gasclaw.health import _list_agents
+
+        def mock_run(*a, **kw):
+            return subprocess.CompletedProcess(
+                args=a[0] if a else ["cmd"],
+                returncode=0,
+                stdout=b"   \n  \t  \n \n\n",
+            )
+
+        monkeypatch.setattr(subprocess, "run", mock_run)
+        result = _list_agents()
+        assert result == []
+
+    def test_list_agents_internal_whitespace_preserved(self, monkeypatch):
+        """Test _list_agents preserves internal whitespace in agent names."""
+        from gasclaw.health import _list_agents
+
+        def mock_run(*a, **kw):
+            return subprocess.CompletedProcess(
+                args=a[0] if a else ["cmd"],
+                returncode=0,
+                stdout=b"crew worker 1\ncrew-worker-2\n",
+            )
+
+        monkeypatch.setattr(subprocess, "run", mock_run)
+        result = _list_agents()
+        # Internal whitespace should be preserved
+        assert result == ["crew worker 1", "crew-worker-2"]
+
 
 class TestHealthReportSummary:
     def test_summary_string(self):

